### PR TITLE
ACS-5975 change getting tasks in tasklist component and service

### PR DESCRIPTION
**Pull request by DominikIwanek**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/ACS-5975
If ‘all’ state filter is selected, tasklist component’s loadTasksByState method calls findAllTaskWithoutState method in service, that sends three requests

**What is the new behaviour?**
Only one request is sent, there are no extra requests

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/process-services/src/lib/task-list/services/tasklist.service.ts [hunk 1]
```diff
@@ -124,10 +124,10 @@ export class TaskListService {
      * Gets all tasks matching a query and state value.
      *
      * @param requestNode Query to search for tasks.
-     * @param state Task state. Can be "open" or "completed".
+     * @param state Task state. Can be "open" or "completed". State "all" applies to both "active" and "completed" tasks.
```
Comment (AleksanderSklorz) [inline lib/process-services/src/lib/task-list/services/tasklist.service.ts:127]: "State "all" applies to both "**active**" and "completed" tasks." -  is that correct? Statement before this statement says: "Can be "**open**" or "completed"."

File: lib/process-services/src/lib/mock/task/tasklist-service.mock.ts [hunk 2]
```diff
@@ -171,17 +171,3 @@ export const fakeTaskCompleted2 = {
     dueDate: null,
     endDate: '2016-11-03T15:25:42.749+0000'
 };
-
-export const fakeOpenTaskList = new TaskListModel({
-    size: 2,
-    total: 2,
-    start: 0,
-    data: [fakeTaskOpen1, fakeTaskOpen2]
```
Comment (AleksanderSklorz) [inline lib/process-services/src/lib/mock/task/tasklist-service.mock.ts:179]: I think you can remove now fakeTaskOpen1 and fakeTaskOpen2 as they are unused.

File: lib/process-services/src/lib/mock/task/tasklist-service.mock.ts [hunk 3]
```diff
@@ -171,17 +171,3 @@ export const fakeTaskCompleted2 = {
...
-});
-
-export const fakeCompletedTaskList = new TaskListModel({
-    size: 2,
-    total: 2,
-    start: 0,
-    data: [fakeTaskCompleted1, fakeTaskCompleted2]
```
Comment (AleksanderSklorz) [inline lib/process-services/src/lib/mock/task/tasklist-service.mock.ts:186]: I think you can remove now fakeTaskCompleted1 and fakeTaskCompleted2 as they are unused.

**Review by AleksanderSklorz — APPROVED** [review]

File: lib/process-services/src/lib/task-list/services/tasklist.service.ts [hunk 4]
```diff
@@ -104,7 +104,7 @@ export class TaskListService {
      * Gets tasks matching a query and state value.
      *
      * @param requestNode Query to search for tasks
-     * @param state Task state. Can be "open" or "completed".
+     * @param state Task state. Can be "open" or "completed". State "all" applies to both "open" and "completed" tasks.
```
Comment (swapnil-verma-gl) [inline lib/process-services/src/lib/task-list/services/tasklist.service.ts:107]: The documentation file defines the state parameter as 
> Task state. Can be **"active"** or "completed". State "all" applies to both "active" and "completed" tasks.

Whereas in code, it is defined as 
> Task state. Can be **"open"** or "completed". State "all" applies to both "open" and "completed" tasks.

**Pull request merged**